[Caffe2][Slice] Extend Caffe2 Slice to support axes argument

Fiery opened the pull request:
Summary: Extends caffe2 slice operator to support an optional axes argument.

Differential Revision: D19544031

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit aede33d5:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=aede33d57be4ce93cbe51e0e50bfdd85702ff75b).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4364378) pytorch_linux_xenial_py3_6_gcc5_4_ge_config_legacy_test (1/1)
**Step:** "Set Up CI Environment After attach_workspace" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=92243442) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=92243442)) :snowflake:
<details>
<summary>
<code>gpg: no valid OpenPGP data found.</code>
</summary>

```
+ curl -s -L --retry 3 https://nvidia.github.io/nvidia-docker/gpgkey 
+ sudo apt-key add - 
gpg: no valid OpenPGP data found. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
[Caffe2][Slice] Extend Caffe2 Slice to support axes argument (#32737)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32737

Extends caffe2 slice operator to support an optional axes argument.

Differential Revision: D19544031

fbshipit-source-id: cf2e432f898447b39b133c4b2df23f6ff836f689
